ngraph/core/src/type.cpp

```diff
@@ -4,20 +4,47 @@
 
 #include "ngraph/type.hpp"
 
-#include "ngraph/util.hpp"
+#include "openvino/util/common_util.hpp"
 
 namespace std {
 size_t std::hash<ngraph::DiscreteTypeInfo>::operator()(const ngraph::DiscreteTypeInfo& k) const {
-    NGRAPH_SUPPRESS_DEPRECATED_START
-    size_t name_hash = hash<string>()(string(k.name));
-    size_t version_hash = hash<decltype(k.version)>()(k.version);
-    // don't use parent for hash calculation, it is not a part of type (yet)
-    return ngraph::hash_combine(vector<size_t>{name_hash, version_hash});
-    NGRAPH_SUPPRESS_DEPRECATED_END
+    return k.hash();
 }
 }  // namespace std
 
 namespace ov {
+
+size_t DiscreteTypeInfo::hash() const {
+    if (hash_value != 0)
```

Comment:
what if `hash_value` is computed as `0`? 😄

Reply:
So, it means that we will have a little performance degradation for this particular case. But any way, it is not a big impact.